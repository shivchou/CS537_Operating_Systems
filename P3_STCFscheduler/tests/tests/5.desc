give_cpu test: ensure give_cpu allows other runnable process to run under STCF.
