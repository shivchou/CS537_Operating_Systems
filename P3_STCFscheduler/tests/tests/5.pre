./edit-makefile.sh ../solution/Makefile test_5 > ../solution/Makefile.test
cp -f tests/test_5.c ../solution/test_5.c
cd ../solution/
make clean
make -f Makefile.test clean
make -f Makefile.test xv6.img
make -f Makefile.test fs.img
cd ../tests
