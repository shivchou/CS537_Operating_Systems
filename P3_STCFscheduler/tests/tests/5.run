cd ../solution; ../tests/run-xv6-command.exp CPUS=1 SCHEDULER=STCF Makefile.test test_5 | grep -E 'XV6_TEST'; cd ../tests
