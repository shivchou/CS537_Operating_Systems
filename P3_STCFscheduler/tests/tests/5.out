XV6_TEST: scheduler : STCF
XV6_TEST: give_cpu test started
XV6_TEST: child started
XV6_TEST: scheduled to parent
XV6_TEST: child finished
XV6_TEST: give_cpu test fin
